src/com/model/BanknotesHistory.java

```diff
@@ -0,0 +1,49 @@
+package com.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BanknotesHistory {
+    public Map<Integer, BanknoteHistoryData> banknotesHistoryMap = new HashMap<>();
+
+    // Init historyMap
+    // key - banknoteNominal
+    // value - BanknoteHistoryData class
+    BanknotesHistory(int[] allowedBanknotes) {
+        for (int banknoteNominal : allowedBanknotes) {
```

Comment:
Initializing the "whole" hashmap is not actually needed here. Because when you refer to a map entry for the first time, you can use `banknotesMap.putIfAbsent(nominal, new HistoryData()).setBanknotesReceived(...)`
Or even, if you don't do it this way, anyway, init the map entry when it is actually needed, and not for "just letting it be")